### PR TITLE
feat(makefile): introduce local-env to run local aurora & ipfs

### PR DESCRIPTION
**Pull request by folex**
## Description
Introduce `local-env` and `local-nox` to Makefile

- `local-env` runs aurora and ipfs containers
- `local-nox` runs Nox via `cargo run`, configured to use RPC & IPFS from `local-env`

## Motivation
Make it easier to test deal deploys and related things.

## Proposed Changes
- Add `local-env` command to Makefile
- Add `local-nox` command to Makefile
- Introduce `enable_logs_for` to make it possible to set log levels via function call. Very useful if use `created-swarms` & `log-utils` as crates in some other project

## Checklist
- [ ] The code follows the project's coding conventions and style guidelines.
- [ ] All tests related to the changes have passed successfully.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ ] All new and existing unit tests have passed.
- [ ] I have self-reviewed my code and ensured its quality.
- [ ] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Code has been reviewed for quality and adherence to [guidelines](https://doc.rust-lang.org/1.0.0/style/README.html).
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: crates/server-config/src/dir_config.rs [hunk 1]
```diff
@@ -29,26 +29,20 @@ pub struct UnresolvedDirConfig {
     pub base_dir: PathBuf,
 
     /// Base directory for resources needed by application services
-    #[serde(default)]
```
Comment (justprosh) [inline crates/server-config/src/dir_config.rs:32]: Why?

**Review by gurinderu — COMMENTED** [review]
1. Why do we need a make file? You can probably use a compose file directly
2. If we still need a make file why is it not described in README.md?
3. Why have you made unrelated things in this pr? (I mean you touched config)

File: crates/server-config/src/dir_config.rs (hunk 1 above)
Comment (folex) [reply crates/server-config/src/dir_config.rs:32]: Because it works without it. Why do we need it then?

**Comment by folex** [comment]
> * Why do we need a make file? You can probably use a compose file directly
It's like scripts in package.json in JS/TS. Very useful, gives you way to share commands with fellow contributors. Good entrypoint.

Also, makefile somehow makes it easier for Cargo to handle incremental compilation. I guess it isolates environment somehow, and Cargo feels better.

> * If we still need a make file why is it not described in README.md?

Would be nice to describe it, I guess.

> * Why have you made unrelated things in this pr? (I mean you touched config)

Why not? It's a few lines. Not sure moving it to a separate PR is worth the overhead.

p.s. comments to code lines are much easier to reply to, on github.

**Review by gurinderu — APPROVED** [review]

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  fluence:
+    driver: bridge
+    ipam:
+      config:
+        - subnet: 10.50.10.0/24
+
+services:
+  aurora:
+    image: docker.fluence.dev/deal-aurora:0.2.4
+#    image: docker.fluence.dev/aurora:main-9e7523f-4-1
```
Comment (justprosh) [inline docker-compose.yml:11]:
```suggestion
```

File: aquamarine/src/aqua_runtime.rs [hunk 3]
```diff
@@ -128,7 +128,7 @@ impl AquaRuntime for AVM<DataStoreError> {
                 }
             }
             Ok((data, ..)) => {
-                tracing::warn!(
+                tracing::info!(
```
[the remark this answers is not in the document]
Comment (folex) [reply aquamarine/src/aqua_runtime.rs:131]: because it's a super normal situation. Might even move it to DEBUG.

**Comment by kmd-fl** [comment]
Envs doesn't work properly. Nox prints
```
ignoring `	aquamarine::aqua_runtime=off`: invalid filter directive
... 10 similar entries ...
ignoring `	particle_reap=off`: invalid filter directive
```

**Comment by kmd-fl** [comment]
If I remove tabs (or even spaces) from strings, it works fine
```
server:
	WASM_LOG="trace" \
	RUST_LOG="info,\
aquamarine::aqua_runtime=off,\
ipfs_effector=off,\
ipfs_pure=off,\
system_services=debug,\
marine_core::module::marine_module=info,\
aquamarine=warn,\
tokio_threadpool=info,\
tokio_reactor=info,\
mio=info,\
tokio_io=info,\
soketto=info,\
yamux=info,\
multistream_select=info,\
libp2p_secio=info,\
libp2p_websocket::framed=info,\
libp2p_ping=info,\
libp2p_core::upgrade::apply=info,\
libp2p_kad::kbucket=info,\
cranelift_codegen=info,\
wasmer_wasi=info,\
cranelift_codegen=info,\
wasmer_wasi=info,\
run-console=trace,\
wasmtime_cranelift=off,\
wasmtime_jit=off,\
particle_reap=off" \
	cargo run --release -p nox
```

**Pull request merged**